pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -39,12 +40,66 @@ func ApplyClusterMerge(pctx networking.EnvoyFilter_PatchContext, efw *model.Envo
 			continue
 		}
 		if commonConditionMatch(pctx, cp) && clusterMatch(c, cp) {
-			proto.Merge(c, cp.Value)
+
+			ret, err := mergeTransportSocketCluster(c, cp)
+			if err != nil {
+				continue
+			}
+			if !ret {
+				proto.Merge(c, cp.Value)
+			}
 		}
 	}
 	return c
 }
 
+// Test if the patch contains a config for TransportSocket
+func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfigPatchWrapper) (bool, error) {
+
+	err := fmt.Errorf("ERROR function mergeTransportSocketCluster failed.")
+
+	cpValueCast, okCpCast := (cp.Value).(*cluster.Cluster)
+	if !okCpCast {
+		log.Errorf("ERROR Cast of cp.Value: %v", okCpCast)
```

Comment:
I think we avoid logging in this code as it is called 1000s of times and can overload logs.

Also, I don't think this can actually happen